apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -632,4 +652,34 @@ defmodule Aecore.Chain.Worker do
   end
 
   defp build_chain_state, do: Chainstate.init()
+
+  def transfrom_chainstate(strategy, chainstate) do
```

Comment:
Why exactly transform? 
We transform the chainstate state to be in a suitable shell for read and write and depending on whether we want to read or write the state, we have these two strategies which we choose from.

Reply:
can you describe it as comment in code?

Reply:
@DanielaIvanova can you add this as well?

Reply:
@thepiwo because we persist only the root hashes of all chain states we need to transform those chain states to one holding only the root hashes. When we start the node we have to transform persisted chain states from root hashes to normal chain states tries

Reply:
sorry I don't understand how you transform a hash to data

Reply:
We should store whole all chainstate data in Persistance not only root hashes of trees; I also don't understand how you transform a hash to data.

Reply:
@thepiwo @cytadela8  we are constructing patricia tries from previous root hashes, so we do not need to persist anything but the root hashes. Patricia merkle library persist this in rox (in our case) implicitly for us.

Reply:
can you explain this further? how can we know the data, e.g. one account balance after restarting the node.

Reply:
@cytadela8 the data leaves of patricia merkle trees are persisted using the reference, this only refers how to store the state tree with these references.